packages/common/src/api/tan-query/wallets/useUserCoinBalance.ts

```diff
@@ -0,0 +1,88 @@
+import { useQuery } from '@tanstack/react-query'
+
+import { useCurrentUserId } from '~/api/tan-query/users/account/useCurrentUserId'
+import { ID } from '~/models/Identifiers'
+
+import { QUERY_KEYS } from '../queryKeys'
+import { QueryKey, SelectableQueryOptions } from '../types'
+
+type UserCoinBalanceResponse = {
+  data: Array<{
+    owner: string
+    account: string
+    balance: number
+  }>
+}
+
+export const getUserCoinBalanceQueryKey = (
+  userId: ID | null | undefined,
+  mint: string | null | undefined
+) =>
+  [
+    QUERY_KEYS.userCoinBalance,
+    userId,
+    mint
+  ] as unknown as QueryKey<UserCoinBalanceResponse | null>
+
+type UseUserCoinBalanceParams = {
+  userId: ID | null | undefined
+  mint: string
+}
+
+/**
+ * Hook to get a user's balance for a specific coin mint address.
+ * Currently mocked since the SDK method is not yet implemented.
+ * For the current user, always returns a balance.
+ * For other users, returns a balance 50% of the time.
+ */
+export const useUserCoinBalance = <TResult = UserCoinBalanceResponse | null>(
```

Comment:
I think we should name this something like `useTokenBalance` or something. i actually have a hook for this in my upcoming PR 😅 

https://github.com/AudiusProject/audius-protocol/blob/9c4a513e9cfece1fa711a63d583a11f70afae87e/packages/common/src/api/tan-query/wallets/useTokenBalance.ts

Reply:
oh whoops. how do u wanna handle? should i just wait for your PR to merge and then add randomness there?